### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by kateryna3k**
Screenshot 2022-10-31 at 20 30 14
Screenshot 2022-10-31 at 20 30 22
Screenshot 2022-10-31 at 20 30 30
Screenshot 2022-10-31 at 20 30 37
Screenshot 2022-10-31 at 20 30 44
Screenshot 2022-10-31 at 20 31 00
Screenshot 2022-10-31 at 20 31 07

### CODE REVIEW

File: taxi/admin.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from django.contrib import admin
+from django.contrib.auth.admin import UserAdmin
+
+from taxi.models import Car, Driver, Manufacturer
+
+
+@admin.register(Car)
+class CarAdmin(admin.ModelAdmin):
+    search_fields = ["model"]
+    list_filter = ["manufacturer"]
```
Comment (mltdll) [inline taxi/admin.py:10]: Looks like your screenshots are outdated. -_-

File: taxi/models.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def __str__(self):
+        return self.name
+
+
+class Driver(AbstractUser):
+    license_number = models.CharField(max_length=255, unique=True)
+
+    class Meta:
+        verbose_name = "Driver"
+        verbose_name_plural = "Drivers"
```
Comment (mltdll) [inline taxi/models.py:18]: I guess these names should be lowercase.

File: taxi/models.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from django.db import models
+from django.contrib.auth.models import AbstractUser
+
+
+class Manufacturer(models.Model):
+    name = models.CharField(max_length=255, unique=True)
+    country = models.CharField(max_length=255)
+
+    def __str__(self):
```
Comment (mltdll) [inline taxi/models.py:9]: What about the type annotations?

**Review by mltdll — CHANGES_REQUESTED** [review]

File: taxi/models.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        return self.username
+
+
+class Car(models.Model):
+    model = models.CharField(max_length=255)
+    manufacturer = models.ForeignKey(
+        Manufacturer, on_delete=models.CASCADE
+    )
+    drivers = models.ManyToManyField(
+        Driver, related_name="cars"
```
Comment (mltdll) [inline taxi/models.py:30]: use `settings.AUTH_USER_MODEL`.

**Review by mltdll — CHANGES_REQUESTED** [review]

File: taxi/admin.py (hunk 1 above)
Comment (kateryna3k) [reply taxi/admin.py:10]: I've checked recently and nothing changed

Comment (mltdll) [reply taxi/admin.py:10]: There should be a filter field in the car list. If there isn't, it means that there's something wrong.

Comment (kateryna3k) [reply taxi/admin.py:10]: I understand, but I don't understand, why there is no this field
Thanks, I'll try to do smth with it

Comment (kateryna3k) [reply taxi/admin.py:10]: Everything is ok)
I just added a second manufacturer)))))
Screenshot 2022-11-01 at 12 27 47

**Review by mltdll — APPROVED** [review]

**Review by TPolina — APPROVED** [review]
Well done!

**Pull request closed without merging**